File: Labs/Lab-04-20231013/app/src/main/java/sap/escooters/presentation_layer/PresentationLayerImpl.java
package sap.escooters.presentation_layer;

import java.util.Optional;

import io.vertx.core.Vertx;
import sap.escooters.service_layer.*;
import sap.layers.Layer;

public class PresentationLayerImpl implements PresentationLayer {

	private ServiceLayer serviceLayer;
	
	public PresentationLayerImpl() {	
	}
		
	@Override
	public void init(Optional<Layer> layer) {
		serviceLayer = (ServiceLayer) layer.get();
    	Vertx vertx = Vertx.vertx();
		EScooterManServer myVerticle = new EScooterManServer(8081, serviceLayer);
		vertx.deployVerticle(myVerticle);		
	}
}

File: Labs/Lab-05-20231016/app/src/main/kotlin/assigment03/escooters/presentation_layer/PresentationLayerImpl.kt
package assigment03.escooters.presentation_layer

import assigment03.escooters.service_layer.ServiceLayer
import assigment03.layers.Layer
import io.vertx.core.Vertx

class PresentationLayerImpl(private val port: Int) : PresentationLayer {

    private lateinit var serviceLayer: ServiceLayer

    override fun init(layer: Layer?) {
        serviceLayer = layer as ServiceLayer
        val vertx = Vertx.vertx()
        val myVerticle = EScooterManServer(port, serviceLayer)
        vertx.deployVerticle(myVerticle)
    }
}